### PR TITLE
Control visibility of each category in logger widget

### PR DESCRIPTION
**Pull request by patrickelectric**
- Log all categories by default
- Control visibility of each category
- Can see old messages of categories that visibility was off 

Needs #308

### CODE REVIEW

File: src/logger/logger.cpp [hunk 1]
```diff
@@ -48,21 +41,22 @@ QObject* Logger::qmlSingletonRegister(QQmlEngine* engine, QJSEngine* scriptEngin
...
     const QString time = QTime::currentTime().toString(QStringLiteral("[hh:mm:ss:zzz]"));
 
     // Save the message into the file
     _fileStream << QString("%1 %2\n").arg(time, msg);
 
-    const int line = _logModel.rowCount();
     // Debug, Warning, Critical, Fatal, Info
     static const QColor colors[] = { QColor("DarkGray"), QColor("orange"), QColor("red"), QColor("red"), QColor("LimeGreen") };
 
+    const int line = _logModel.rowCount();
```
Comment (jaxxzer) [inline src/logger/logger.cpp]: why this move

Comment (patrickelectric) [reply src/logger/logger.cpp]: just to be closer to the usage of the variable.

File: src/logger/logger.cpp [hunk 2]
```diff
@@ -90,33 +84,19 @@ void Logger::registerCategory(const char* category)
     if (_registeredCategories.contains(category)) {
         return;
     }
+    // Register each category in a bit, this will help the qml element to be faster when searching between categories
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply src/logger/logger.cpp]: We are using a QMap to translate category name to index, how can this break ?

[the remark this answers is not in the document]
Comment (jaxxzer) [reply src/logger/logger.cpp]: Of course, I confused myself.

[the remark this answers is not in the document]
Comment (jaxxzer) [reply src/logger/logger.cpp]: The category name <-> index # relationship/binding is not static. Should we make a category enum?

[the remark this answers is not in the document]
Comment (patrickelectric) [reply src/logger/logger.cpp]: Yes, it it's possible, but only after updates of the graphic interface. That's a small price for the functionality though.

**Pull request merged**